src/main/java/org/nypl/findawayclientlibrary/DownloadService.java

```diff
@@ -553,46 +543,75 @@ public void onNext(DownloadEvent downloadEvent) {
         // which would be hard to catch based on a string message, but common enough to need handling.
 
       } else if (DownloadEvent.CHAPTER_ALREADY_DOWNLOADED.equals(downloadEvent.code())) {
-        LogHelper.e(TAG, "DownloadEvent.NOT_ENOUGH_SPACE_ERROR");
+        LogHelper.d(TAG, "DownloadEvent.CHAPTER_ALREADY_DOWNLOADED");
+        // TODO: ungray the chapter in the TOC
+
       } else if (DownloadEvent.CHAPTER_ALREADY_DOWNLOADING.equals(downloadEvent.code())) {
-        LogHelper.e(TAG, "DownloadEvent.NOT_ENOUGH_SPACE_ERROR");
+        LogHelper.d(TAG, "DownloadEvent.CHAPTER_ALREADY_DOWNLOADING");
       }
 
     } else {
       // download event is not an error, whee
       if (downloadEvent.code().equals(DownloadEvent.DOWNLOAD_STARTED)) {
-        // TODO: make sure all string messages are coming in from R.string
-        callbackActivity.notifyDownloadEvent(callbackActivity.getString(R.string.downloadStarted));
-        callbackActivity.setDownloadProgress(0, 0, DOWNLOAD_STATUS.DOWNLOAD_RUNNING);
+        // DOWNLOAD_STARTED gets called for every chapter that is to be downloaded, not just for the whole book.
+
+        // if download is resuming from not the first chapter, then it's ok not to show the toast
+        // if there is an introduction, it will be labeled as chapter 0, and it's ok not to show the toast until
+        // the first chapter is being downloaded.
+        if (chapter != null && new Integer(1).equals(chapter.chapter())) {
+          callbackActivity.notifyDownloadEvent(callbackActivity.getString(R.string.downloadStarted));
+        }
+        //callbackActivity.drawDownloadProgressButton(downloadEvent.contentPercentage(), downloadEvent.chapterPercentage(), DOWNLOAD_RUNNING);
```

Comment:
There are a lot of lines in this file that are commented out. If we do not need them, can we just get rid of them?

Reply:
yes